ObjCVideoQuickstart/ViewController.m

```diff
@@ -0,0 +1,314 @@
+                if (!err) {
+                    self.accessToken = token;
+                } else {
+                    [self logMessage:[NSString stringWithFormat:@"Error retrieving the access token"]];
+                }
+            });
+        }];
+    }
+    
+    // LocalMedia represents the collection of tracks that we are sending to other Participants from our VideoClient.
+    self.localMedia = [[TVILocalMedia alloc] init];
+    
+    if ([PlatformUtils isSimulator]) {
+        [self.previewView removeFromSuperview];
+    } else {
+        // Preview our local camera track in the local video preview view.
+        [self startPreview];
+    }
+    
+    // Disconnect and mic button will be displayed when client is connected to a room.
+    self.disconnectButton.hidden = YES;
+    self.micButton.hidden = YES;
+    
+    self.roomTextField.delegate = self;
+
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(dismissKeyBoard)];
+    [self.view addGestureRecognizer:tap];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+- (void)startPreview {
+    if ([PlatformUtils isSimulator]) {
+        return;
+    }
+    
+    self.camera = [[TVICameraCapturer alloc] init];
+    self.localVideoTrack = [self.localMedia addVideoTrack:YES capturer:self.camera];
+    if (!self.localVideoTrack) {
+        [self logMessage:@"Failed to add video track"];
+    } else {
+        // Attach view to video track for local preview
+        [self.localVideoTrack attach:self.previewView];
+        
+        [self logMessage:@"Video track added to localMedia"];
+        
+        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self
+                                                                              action:@selector(flipCamera)];
+        [self.previewView addGestureRecognizer:tap];
+    }
+}
+
+- (void)flipCamera {
+    [self.camera flipCamera];
+}
+
+- (void)prepareLocalMedia {
+    
+    // We will offer local audio and video when we connect to room.
+    
+    // Adding local audio track to localMedia
+    self.localAudioTrack = [self.localMedia addAudioTrack:YES];
+    
+    // Adding local video track to localMedia and starting local preview if it is not already started.
+    if (self.localMedia.videoTracks.count == 0) {
+        [self startPreview];
+    }
+}
+
+- (IBAction)connectButtonPressed:(id)sender {
+    if([self.accessToken isEqualToString:@"TWILIO_ACCESS_TOKEN"]) {
+        [self logMessage:@"Please provide a valid token to connect to a room."];
+        return;
+    }
+    
+    // Creating a video client with the use of the access token.
+    if (!self.client) {
+        self.client = [TVIVideoClient clientWithToken:self.accessToken];
+        if (!self.client) {
+            [self logMessage:@"Failed to create video client"];
+            return;
+        }
+    }
```

Comment:
Suggestion:
`// Create a Client with the access token that we fetched (or hardcoded).`